docs/v0.5/en/user-guide/operations/kafka_remote_wal.md

```diff
@@ -0,0 +1,358 @@
+
+[Write-Ahead Logging](https://docs.greptime.com/contributor-guide/datanode/wal#introduction) (WAL) is a crucial component in GreptimeDB. Every data modification operation is logged in the WAL to ensure that the database does not lose data cached in memory.
+
+Before version 0.5, we utilized an embedded [Raft Engine](https://www.pingcap.com/blog/raft-engine-a-log-structured-embedded-storage-engine-for-multi-raft-logs-in-tikv/) as the storage engine for WAL. While in practical deployments, we could mount the Raft Engine on cloud storage to achieve a [Recovery Point Objective](https://en.wikipedia.org/wiki/Disaster_recovery#Recovery_Point_Objective) (RPO) of 0, the remounting process incurred a significant [Recovery Time Objective](https://en.wikipedia.org/wiki/Disaster_recovery#Recovery_Time_Objective) (RTO). Additionally, the embedded Raft Engine couldn't meet the demands of multiple subscriptions to logs, hindering features like [hot standby](https://en.wikipedia.org/wiki/Hot_spare) and region migration in GreptimeDB.
+
+With the release of version [0.5](https://github.com/GreptimeTeam/greptimedb/releases/tag/v0.5.0), we transitioned to using a remote storage service as the WAL storage engine, referred to as Remote WAL. [Apache Kafka](https://kafka.apache.org/), widely adopted in the field of stream processing, exhibits excellent distributed fault tolerance and a subscription mechanism based on [Topics](https://www.conduktor.io/kafka/kafka-topics/). This aligns well with GreptimeDB's current requirements for Remote WAL. Hence, in version 0.5, we introduced Apache Kafka as an optional storage engine for WAL.
+
+# How to use Kafka Remote WAL
+
+## Step 1: Start Kafka Cluster
+
+If you have already deployed a Kafka cluster, you can skip this step. However, please pay attention to the [advertised listeners](https://www.conduktor.io/kafka/kafka-advertised-host-setting/) configured during deployment, as you will use them in Step 2.
+
+
+
+We recommend using [docker compose](https://docs.docker.com/compose/) to start the Kafka cluster. Kafka supports two deployment modes: [KRaft](https://www.conduktor.io/kafka/kafka-kraft-mode/) and [Zookeeper](https://www.conduktor.io/kafka/zookeeper-with-kafka/). You can find docker compose scripts for KRaft and Zookeeper modes [here](https://github.com/confluentinc/kafka-images/blob/master/examples/confluent-server-kraft/docker-compose.yml) and [here](https://github.com/conduktor/kafka-stack-docker-compose) respectively. We advise deploying in KRaft mode, similar to the [docker-compose-standalone.yml](https://github.com/GreptimeTeam/greptimedb/blob/main/tests-integration/fixtures/kafka/docker-compose-standalone.yml) script we use, which is also pasted below for your convenience.
+
+```yaml
+version: '3.8'
+services:
+  kafka:
+    image: bitnami/kafka:3.6.0
+    container_name: kafka
+    ports:
+      - 9092:9092
+    environment:
+      # KRaft settings
+      KAFKA_KRAFT_CLUSTER_ID: Kmp-xkTnSf-WWXhWmiorDg
```

Comment:
Where is the id from? Could users use it directly?

Reply:
Yes, users could use the script directly without worrying about the id.